src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala

```diff
@@ -895,13 +896,14 @@ object OpenSearchResponses {
         .valueOr(e => throw new IllegalArgumentException(s"${e.show} while parsing '$json'", e))
     }
 
-    private def retainTileIdPattern(features: Array[Feature], tileIdPattern: Option[String]): Array[Feature] =
-      tileIdPattern match {
-        case Some(pattern) => features.filter(feature => feature.tileID match {
+    private def retainTileIds(features: Array[Feature], tileIdValue: Option[Any]): Array[Feature] =
```

Comment:
Would it make sense to dedup with this? https://github.com/Open-EO/openeo-opensearch-client/blob/c3e732c8e8966c9d6be365378c0a8954216b10b8/src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala#L131